File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int n = 0;
        
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)=='('){
                n++;
            }else{
                n--;
            }
            if (n<0){
                return false;
            }
        }
        
        if(n!=0){
            return false;
        }else{
            return true;
        }
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;

    const stack = [];
    
    for(let i of s){
        if(i=='('){
            stack.push(')')
        }else{
            // if(stack.length>0 && stack[stack.length-1]){
            if(stack.length>0){
                stack.pop();
            }else{
                return false;
            }
        }
    }
    
    if(stack.length>0){
        return false;
    }
    
    return answer;
}